### ISSUE TITLE
Generator Wizard Options

### ISSUE THREAD

**Issue by secondmanveran** [shomisha/laravel-console-wizard#4]
Okay, I have a genuine issue now that I'd like to figure out how to resolve.

All of the wizards in my package will be generators. So I've discovered I need to extend the `GeneratorWizard` class for this. Most of them will work fine as is since they are creating a named file. But I have a couple that don't create an actual class, they just call other generator commands.

For instance my `ScaffoldMakeCommand` I referenced in that previous issue. I moved the the first step into the `getNameStep` which works as expected, then moved my build calls from `completed` into `generateTarget` and those all run and generate all the files as expected.

The one exception here is that an extra empty file is created based on the input from `getNameStep`.

For example, here is my `getNameStep`:

```php
public function getNameStep(): Step
{
  return new TextStep('What namespace would you like to create?');
}
```

This sets a namespace to then generate a bunch of files within that directory. So let's say I want to create a `Countries` namespace. 

I go through the wizard answering all the steps:

```php
public function getSteps(): array
{
  return [
    'visibility' => new ChoiceStep('What visibility should the namespace have?', [
      'Public', 'Protected', 'Private', 'API',
    ]),
    'actions' => new UniqueMultipleChoiceStep('Select all the actions to create for your namespace', [
      'Index', 'Create', 'Show', 'Store', 'Edit', 'Update', 'Delete', 'Restore', 'Destroy',
    ], [
      'end_keyword' => 'done',
      'retain_end_keyword' => false,
    ]),
    'model' => new TextStep('Enter the name of the Model to use for this namespace'),
    'repository' => new ConfirmStep('Would you like a repository?'),
    'observer' => new ConfirmStep('Do you need an Observer?'),
  ];
}
```

Then all the files requested are created, and at the end I get an empty file at:

```php
app/Countries.php
```

Is there a way I can stop this file from being created? In my previous command which just extended `GeneratorCommand` this did not get created because I left the `getStub` method empty. I did that here too but it still gets created, I'm guessing maybe from the `parent::handle()` call in `GeneratorWizard`?

Any thoughts would be helpful.

Thanks again for the great package.

**Comment by shomisha** [comment]
Hey again :) 

There's not really an easy way to achieve this with the `GeneratorWizard` abstract. It inherits Laravel's native `Illuminate\Console\GeneratorCommand` which stores the newly generated file on the disk directly in its `handle()` method, so there is no method you can override to stop this from happening since the wizard calls that method (as you said, it's in the `parent::handle()` call). 

The `getStub()` method in the `GeneratorWizard` is also empty and it's `final`, so there's not a way nor a need to override it as it already returns null.

Basically, I think the easiest way to get what you want is not to use a `GeneratorWizard` because it's intended for creating single files (you already have to add some custom logic for creating other files) but extend the regular `Shomisha\LaravelConsoleWizard\Command\Wizard` abstract which let's you do whatever you want with the input you've received

**Comment by shomisha** [comment]
Hey, thanks for getting back with this. I don't really see this as a broadly needed use-case, I know there are packages that scaffold a lot of code but I think there are other ways of implementing this. Also, it's mixing a few different concerns so I don't really think it would be simple and straightforward to merge it into the package.

If you need to run other generators from the main one did you consider an approach in which the main wizard is a plain wizard that simply runs other generators based on the input and doesn't generate a file on its own? This way you wouldn't have to override methods or do any exhibitions, the wizard would be the wizard and the generators would be generators 

Disclaimer: since I haven't had an insight into your code I'm just making suggestions based on what I read from your comments. There could be limitations or other factors I am unaware of
